src/lib/utils/convertPercentageColor.ts

```diff
@@ -0,0 +1,47 @@
+import { ColorValue } from 'react-native';
+
+export const RGB_PATTERN =
+  /^rgb\(\s*(100%|100|\d{1,2}(\.\d+)?%)\s*,\s*(100%|100|\d{1,2}(\.\d+)?%)\s*,\s*(100%|100|\d{1,2}(\.\d+)?%)\s*\)$/;
+export const RGBA_PATTERN =
+  /^rgba\(\s*(100%|100|\d{1,2}(\.\d+)?%)\s*,\s*(100%|100|\d{1,2}(\.\d+)?%)\s*,\s*(100%|100|\d{1,2}(\.\d+)?%)\s*,\s*(1|0(\.\d+)?|100%|100|\d{1,2}(\.\d+)?%)\s*\)$/;
```

Comment:
Why do you match 100 as an exact string?
`rgb(100, 100, 100)` should definitely be interpreted as `#646464` rather than `#FFFFFF`

Reply:
[1fbdbb1](https://github.com/software-mansion/react-native-svg/pull/2363/commits/1fbdbb183f627c7c963898044a8d4fdacbf76cc5)